src/math/src/ed25519.cairo

```diff
@@ -1,40 +1,40 @@
-use alexandria_data_structures::array_ext::{ArrayTraitExt, SpanTraitExt};
+use alexandria_data_structures::array_ext::SpanTraitExt;
 use alexandria_math::mod_arithmetics::{
     add_mod, sub_mod, mult_mod, div_mod, pow_mod, add_inverse_mod, equality_mod
 };
 use alexandria_math::sha512::{sha512, SHA512_LEN};
-use core::array::SpanTrait;
-use core::box::BoxTrait;
-use integer::u512;
+use core::integer::u512;
+use core::traits::TryInto;
 
 // As per RFC-8032: https://datatracker.ietf.org/doc/html/rfc8032#section-5.1.7
 // Variable namings in this function refer to naming in the RFC
 
-const p: u256 =
+pub const p: u256 =
     57896044618658097711785492504343953926634992332820282019728792003956564819949; // 2^255 - 19
 const d: u256 =
     37095705934669439343138083508754565189542113879843219016388785533085940283555; // d of Edwards255519, i.e. -121665/121666
 const l: u256 =
     7237005577332262213973186563042994240857116359379907606001950938285454250989; // 2^252 + 27742317777372353535851937790883648493
 
 const TWO_POW_8: u256 = 0x100;
+const TWO_POW_8_NON_ZERO: NonZero<u256> = 0x100;
 
 
 #[derive(Drop, Copy)]
-struct Point {
-    x: u256,
-    y: u256
+pub struct Point {
+    pub x: u256,
+    pub y: u256
 }
 
 #[derive(Drop, Copy)]
-struct ExtendedHomogeneousPoint {
-    X: u256,
-    Y: u256,
-    Z: u256,
-    T: u256,
+pub struct ExtendedHomogeneousPoint {
+    pub X: u256,
+    pub Y: u256,
+    pub Z: u256,
```

Comment:
getters

Reply:
leaving the fields as `pub` is very bad generally speaking

Reply:
laughs in 
https://github.com/starkware-libs/cairo/blob/997a731456d467252086bedf56d1d574d7433eed/corelib/src/integer.cairo#L1320

Reply:
they didn't add a `new` function so it's the only way to create it :'( but it's bad. Also if the corelib already has it why do we define it again ?

Reply:
Done then